frontend/dockerfile/linter/ruleset.go

```diff
@@ -132,4 +132,12 @@ var (
 			return fmt.Sprintf("Setting platform to predefined %s in FROM is redundant as this is the default behavior", platformVar)
 		},
 	}
+	RuleSecretsUsedInArgOrEnv = LinterRule[func(string) string]{
+		Name:        "SecretsUsedInArgOrEnv",
+		Description: "Potentially sensitive data should not be used in the ARG or ENV commands",
+		URL:         "https://docs.docker.com/go/dockerfile/rule/secrets-used-in-arg-or-env/",
+		Format: func(secretKey string) string {
+			return fmt.Sprintf("Secrets should not be used in the ARG or ENV commands (key named %q)", secretKey)
+		},
```

Comment:
Could we pass the instruction type to this rule?

```suggestion
		Format: func(secretKey, kwd string) string {
			return fmt.Sprintf("Do not use ARG or ENV instructions for sensitive data (%s %q)", kwd, secretKey)
		},
```

Reply:
Coming right up!

Reply:
[20224f5](https://github.com/moby/buildkit/pull/5105/commits/20224f5f4615c2a8e16b3225558db45937d389fd)